ui/entity/EntityDeleteConfirmationModal.tsx

```diff
@@ -0,0 +1,67 @@
+import type { UseQueryResult } from '@tanstack/react-query';
+import type { Hex } from 'golem-base-sdk';
+import { useRouter } from 'next/router';
+import React, { useCallback } from 'react';
+
+import type { FullEntity } from '@golembase/l3-indexer-types';
+
+import type { ResourceError } from 'lib/api/resources';
+import { useGolemBaseClient } from 'lib/golemBase/useGolemBaseClient';
+import { toaster } from 'toolkit/chakra/toaster';
+import DeleteModal from 'ui/shared/DeleteModal';
+
+interface Props {
+  entityQuery: UseQueryResult<FullEntity, ResourceError<unknown>>;
+  open: boolean;
+  onOpenChange: ({ open }: { open: boolean }) => void;
+  handleClose: () => void;
+  handleClosePopover: () => void;
+}
+
+const EntityDeleteConfirmationModal = ({
+  entityQuery,
+  open,
+  onOpenChange,
+  handleClose,
+  handleClosePopover,
+}: Props) => {
+  const key = entityQuery.data!.key;
+
+  const router = useRouter();
+  const { createClient } = useGolemBaseClient();
+
+  const handleDelete = useCallback(async() => {
```

Comment:
For consistency with the rest of the app
```suggestion
  const handleDelete = React.useCallback(async() => {
```